src/librados/librados_c.cc

```diff
@@ -2905,6 +2905,75 @@ extern "C" int _rados_notify2(rados_ioctx_t io, const char *o,
 }
 LIBRADOS_C_API_BASE_DEFAULT(rados_notify2);
 
+extern "C" int _rados_decode_notify_response(char *reply_buffer, size_t reply_buffer_len,
+                                             struct notify_ack_t **acks, size_t *nr_acks,
+                                             struct notify_timeout_t **timeouts, size_t *nr_timeouts) {
+  if (!reply_buffer || !reply_buffer_len) {
+    return -EINVAL;
+  }
+
+  bufferlist bl;
+  bl.append(reply_buffer, reply_buffer_len);
+
+  map<pair<uint64_t,uint64_t>,bufferlist> acked;
+  set<pair<uint64_t,uint64_t>> missed;
+  auto iter = bl.cbegin();
+  decode(acked, iter);
+  decode(missed, iter);
+
+  *acks = nullptr;
+  *nr_acks = acked.size();
+  if (acked.size()) {
```

Comment:
nit, if we don't care about the exact size, use `!acked.empty()` please.

Reply:
used `if(*nr_acks)`